engines/experiment/examples/plugins/nop/runner.go

```diff
@@ -0,0 +1,23 @@
+package nop
```

Comment:
>Is this how we are expected to build the binary for the plugin

Building a plugin's binary is no different compared to building any other Go app. So something like this will do the job:
```shell
go build -o bin/nop ./cmd
```

Developer docs are out of scope for now and will be updated once we settle with all the changes to the manager/runner interfaces.